framework/auth/views.py

```diff
@@ -199,10 +199,10 @@ def auth_logout(redirect_url=None):
     return resp
 
 
-def auth_email_logout(token, user):
+def auth_email_logout(token, user, redirect_url=None):
     """When a user is adding an email or merging an account, add the email to the user and log them out.
     """
-    redirect_url = web_url_for('auth_login')
+    redirect_url = cas.get_login_url(redirect_url)
```

Comment:
`redirect_url` will always be `None` here, since you're never passing it to `auth_email_logout`. I don't think `get_login_url` handles `None` as a service URL. Look into other usages of `get_login_url` to figure out the right thing to do here.